core/google/cloud/iterator.py

```diff
@@ -242,7 +242,8 @@ def _page_iter(self, increment):
                           results per page while an items iterator will want
                           to increment per item.
 
-        Yields :class:`Page` instances.
+        :rtype: :class:`Page`
+        :returns: pages
```

Comment:
I have a (maybe not legitimate?) worry here: During a re-factor we may just blindly convert it into a `Returns` section instead of a `Yields` section.